### PR TITLE
nushell: add `--workspace` argument to `cargo`

### PR DESCRIPTION
**Pull request by pthariensflame**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This aligns our formula with the [current public README](https://github.com/nushell/nushell/blob/b3b3cf0689c696327fb6f2f0c560ddcf130c497b/README.md#local).

### CODE REVIEW

**Comment by carlocab** [comment]
What does the `--workspace` flag do?

**Comment by pthariensflame** [comment]
> What does the `--workspace` flag do?

[Here's an explanation of the flag's effect.](https://doc.rust-lang.org/cargo/reference/workspaces.html#package-selection)  In the present case it's merely making the implied default explicit, but that might become required if Nushell later reorganizes its repo; in any case, given that the README recommends it, I felt it was good practice for Homebrew to use it too.

File: Formula/nushell.rb [hunk 1]
```diff
@@ -4,6 +4,7 @@ class Nushell < Formula
   url "https://github.com/nushell/nushell/archive/0.39.0.tar.gz"
   sha256 "86eb8079b88896dfbfb8e165adba75a7de3477b21f20ee8282efff2f0ab1356b"
   license "MIT"
+  revision 1
```
Comment (carlocab) [inline Formula/nushell.rb:7]:
```suggestion
```
If `--workspace` just makes the default explicit then this should have no user-facing changes, right? In which case, we can omit the `revision` bump.

**Comment by carlocab** [comment]
```
error: Found argument '--workspace' which wasn't expected, or isn't valid in this context
```

File: Formula/nushell.rb (hunk 1 above)
Comment (pthariensflame) [reply Formula/nushell.rb:7]: Fair enough!

**Comment by pthariensflame** [comment]
Uh…hm.

**Comment by pthariensflame** [comment]
Ah…I understand now.  This whole PR can be disregarded; the README recommends `--workspace` for `cargo build`, but we're using `cargo install`.  My mistake!

**Pull request closed without merging**